src/components/plots/abstractPlot.ts

```diff
@@ -445,6 +451,43 @@ export module Plot {
 
       return d3.selectAll(allSelections);
     }
+
+    /**
+     * Retrieves the closest PlotData to the specified x/y point within a specified value
+     *
+     * @param {number} xValue The x value to compare against
+     * @param {number} yValue The y value to compare against
+     * @param {number} withinValue The maximum distance the closest selection can be to the point (default = Infinity)
+     * @returns {PlotData} The closest plot data to the point within a specified value.  nulls and null selection returned otherwise
+     */
+    public getClosestData(xValue: number, yValue: number, withinValue: number = Infinity): PlotData {
+      var queryPoint = {x: xValue, y: yValue};
+
+      var closestDatum: any = null;
+      var closestSelection = d3.select();
+      var closestPixelPoint: Point = null;
+      var closestPointDistance = withinValue;
+
+      this._getDrawersInOrder().forEach((drawer) => {
+        drawer._getRenderArea().selectAll(drawer._getSelector()).each(function(datum: any, index: number) {
+          var selection = d3.select(this);
+          var pixelPoints = (drawer instanceof _Drawer.Line) ?
```

Comment:
What happens in a situation where an arbitrary plot uses a Line Drawer?  This case would only happen on Drawers that use an array as a datum.  Should we special case line plots with the possibility that other plots that use Line Drawers will be handing in an array as a datum will need to override again?